src/System.Memory/tests/Memory/TryGetArray.cs

```diff
@@ -45,5 +45,38 @@ public static void OwnedMemoryTryGetArray()
                 Assert.Equal(array[i], segment.Array[i]);
             }
         }
+        
+        [Fact]
+        public static void TryGetArrayFromEmptyMemory()
+        {
+            int[] array = new int[0];
+            Memory<int> memory = array;
```

Comment:
Can you also add a test case where Memory<T> is default and where it is `Memory<T>.Empty` to cover the exact scenario in the original issue (https://github.com/dotnet/corefx/issues/27412)?